scipy/cluster/vq.py

```diff
@@ -557,9 +557,7 @@ def _kpp(data, k):
             init[i, :] = data[np.random.randint(dims)]
 
         else:
-            D2 = np.array([min(
-                            [np.inner(init[j]-x, init[j]-x) for j in range(i)]
-                            ) for x in data])
+            D2 = (cdist(init[:i,:], data)**2).min(axis=0)
```

Comment:
Can you use `metric='sqeuclidean'` instead of squaring after the fact?

Reply:
Yes, thanks for pointing it out!
I just have implemented it; it works and has similar running times.